backend/data/users.js

```diff
@@ -61,6 +63,205 @@ async function checkUser(email, password) {
+  return tempStudent.firstName + " " + tempStudent.lastName;
+}
+
+//Note: Takes in courseId and studentId as strings. 
+//NOTE: This is the function to add a course to a *TEACHER*
+async function addCourse(courseId, teacherId) {
+  //error check inputs
+  error.str(courseId);
+  error.str(teacherId);
+
+  //retrieve the users collection
+  const userCollection = await users();
+
+  //check if the provided teacherId is a valid ObjectId
+  if(!ObjectId.isValid(teacherId)) throw 'Provided ID not proper format for ObjectID!';
+  //convert the ID string to an ObjectID
+  let objId = ObjectId(teacherId);
+
+  //retrieve the original user information
+  let tempTeacher = await userCollection.findOne({ _id: objId });
+
+  //the above call will result in null if the given ID doesn't exist in the database
+  if(tempTeacher === null) throw 'User with that ID is not in database!';
+
+  //if the teacher is already teaching the respective course, throw an error
+  for(let course of tempTeacher.classes)  {
+    if(course._id == courseId)  throw 'Teacher is already teaching that course!';
+  }
+
+  //create new class Object to be added to the teacher's "classes" array
+  const classInfo = {
+    _id: courseId
+  };
+
+  //push the new class Object to the tempTeacher's classes array
+  tempTeacher.classes.push(classInfo);
+
+  //"push" the updated course info to the same ID in the database
+  const updatedInfo = await userCollection.updateOne(
+      {_id: objId},
+      { $set: tempTeacher}
+  );
+
+  //check that the update succeeded
+  if (updatedInfo.modifiedCount === 0) throw 'Failed to add course to teacher!';
+
+
+  //upon successful add, return the name of the teacher for output
+  return tempTeacher.firstName + " " + tempTeacher.lastName;
+}
+
+//Note: Takes in courseId and userId as strings. 
+async function drop(courseId, userId) {
+  //error check inputs
+  error.str(courseId);
+  error.str(userId);
+
+  //retrieve the users collection
+  const userCollection = await users();
+
+  //check if the provided userId is a valid ObjectId
+  if(!ObjectId.isValid(userId)) throw 'Provided ID not proper format for ObjectID!';
+  //convert the ID string to an ObjectID
+  let objId = ObjectId(userId);
+
+  //retrieve the original user information
+  let tempUser = await userCollection.findOne({ _id: objId });
+
+  //the above call will result in null if the given ID doesn't exist in the database
+  if(tempUser === null) throw 'User with that ID is not in database!';
+
+  //Make sure there's at least 1 course in the user's "classes" array
+  if(tempUser.classes.length == 0) throw "'Cannot drop a course from a user that doesn't have any courses!";
+
+  //boolean to track if the course was found in the user's "classes" array
+  let found = false;
+  
+  //find the course in the user's "classes" array
+  for(let course of tempUser.classes)  {
+    //once the course has been found
+    if(course._id == courseId)  {
```

Comment:
Not a deal-breaker but we could enforce type checking by `===` rather than `==`. You just may run into the need of casting the variable into the correct data type. Same logic can be applied to all instances of the use of `==`.